units/commit_queue.go

```diff
@@ -181,6 +183,45 @@ func (j *commitQueueJob) Run(ctx context.Context) {
 			"message": "finished processing item",
 		})
 	}
+	j.AddError(j.addMergeTaskDependencies(*cq))
+}
+
+func (j *commitQueueJob) addMergeTaskDependencies(cq commitqueue.CommitQueue) error {
+	var prevMergeTask string
+	for i, currentItem := range cq.Queue {
+		if currentItem.Version == "" {
+			return nil
+		}
+		mergeTask, err := task.FindOne(task.ByCommitQueueMergeInVersion(currentItem.Version))
+		if err != nil {
+			return errors.Wrap(err, "unable to find merge task")
+		}
+		if mergeTask == nil {
+			return errors.Errorf("no merge task found for version '%s'", currentItem.Version)
+		}
+		dependency := task.Dependency{
```

Comment:
Can we create this dependency after the 208 short circuit? Otherwise it reads as though we would be using this dependency in either case.